hasjob/views/index.py

```diff
@@ -280,32 +238,21 @@ def index(basequery=None, md5sum=None, tag=None, domain=None, location=None, tit
 
     # Pick a header campaign (only if not kiosk or an XHR reload)
     pay_graph_data = None
-    if not g.kiosk:
-        if g.preview_campaign:
-            header_campaign = g.preview_campaign
-        else:
-            geonameids = g.user_geonameids + f_locations
-            header_campaign = Campaign.for_context(CAMPAIGN_POSITION.HEADER, board=g.board, user=g.user,
-                anon_user=g.anon_user, geonameids=geonameids)
-        if pay_graph:
-            pay_graph_data = make_pay_graph(pay_graph, posts, rmin=f_min, rmax=f_max)
-    else:
-        header_campaign = None
 
     loadmore = False
     if batched:
         # Figure out where the batch should start from
         startdate = None
-        if 'startdate' in request.values:
+        if 'startdate' in request_values:
             try:
-                startdate = parse_isoformat(request.values['startdate'])
+                startdate = parse_isoformat(request_values['startdate'])
```

Comment:
We should eliminate `request.values` in this function. If `startdate` is the only use case, let's move it a function parameter. Since `request.values` is a proxy object that looks up both `request.args` and `request.form`, compressing it into a dictionary for passing as an argument to this function is inefficient.